### DISCUSSION TITLE
Regarding documentation hosting

### DISCUSSION THREAD

**Discussion by alkisg** [ltsp/ltsp#201] [Q&A]
This started as an issue to support Raspberry Pis, but then it changed its topic to documentation hosting, so another issue was opened for Raspberry Pis in https://github.com/ltsp/ltsp/issues/68.

**Comment by stevensantos-docs** [comment]
I would be willing to help with this.  I'm not really a coder, but I do have 2 RPi's and a Ubuntu server I can test with.  I also do some linux documentation (user 5g3-steven-7tv on the Ubuntu wiki - I mostly document scanner related things, for example https://help.ubuntu.com/community/Scanners?action=info)  but I have edit access to the ubuntu wiki, and could also do some updates on the ltsp stuff.  I am not going to commit to an entire rewrite of the ltsp constellation over on the ubuntu wiki, and I have limited time to so user hand holding, but I could do the documentation part as time allows.

**Comment by alkisg** [comment]
Hi Steven, thank you for your offer!

I was talking about the LTSP community wiki: https://github.com/ltsp/community/wiki
The LTSP pages in the Ubuntu wiki have been unmaintained for a long time, it's best to completely delete them if you have access there (everything under https://help.ubuntu.com/community/UbuntuLTSP).

I also have a few Raspberries of versions 2, 3 and 4, but I don't have real use for them as LTSP clients. Let's wait a bit in case people that actually need to use Raspberries as LTSP clients e.g. in their classrooms show up; there's no point in doing extra work before the need arises...

**Comment by stevensantos-docs** [comment]
I actually think that using RPi's may become the killer feature for LTSP.  Not for the schools, but for use in clusters.  The thought of being able to add a node by popping in a card and booting... Its appealing. 

Yes, I would be willing to update the ltsp community wiki, not just the ubuntu one.  The ubuntu wiki is the more widely read one, so I think should be updated as well. I'll try to get something done on that in the next week.

**Comment by alkisg** [comment]
This comment was moved to https://github.com/ltsp/ltsp/issues/68#issuecomment-551133173.

**Comment by stevensantos-docs** [comment]
I will run this through a few times and document the process. 

However, it looks like this may all change in a few weeks.  PXE Boot is supported in the latest RPI 4 ROM update beta.  PXE booting should be mainstream available on all RPI devices in just a few weeks.

**Comment by stevensantos-docs** [comment]
https://www.tomshardware.com/features/raspberry-pi-4-firmware-cool-temps-network-boot

**Comment by alkisg** [comment]
Steven may I revert the UbuntuLTSP wiki to what it was, mentioning that it's just no longer used?

https://help.ubuntu.com/community/UbuntuLTSP?action=recall&rev=216

Now it mentions "This is the central hub for all LTSP19 documentation" which is misleading...

**Comment by stevensantos-docs** [comment]
Rewrites don't happen overnight.  One the size of this takes months.

I am happy to work on both the LTSP and Ubuntu documentation as time my
allows, but if you go reverting in-progress work because it moves too slow
for you, I am a lot less likely to contribute any documentation at all.

The LTSP documentation is a friggin mess.   Really cool software, but what
we currently have is a mess.  I have reached the point where I am now
mapping out the documentation that will be needed for LTSP and the Ubuntu
specific version (which as its all deb based at the moment, isn't much
different).

LTSP19 (because having LTSP as the successor to LTSP5 is a documentation
nightmare) and LTSP5 are soo different, the documentation needs to be
seperated. Some things from the old LTSP5 may be recyclable, but that's not
something I would count to heavily on.

So far, in new user documentation we need to:
- Better flesh out the basic concepts for new users.  I started that on the
Ubuntu wiki.
- Identify the basic models (single nic network, dual nic lab, ect) for new
users.
- Identify a few good use-cases for modern LTSP (edu, cluster, media and
retro gaming are the ones I am working on)

Thats the leading documentation.  This is all true of both LTSP5 and LTSP19
and likely any later successors .

LTSP5 documentation needs to be deprecated. To accomplish this, it should
be moved to being under LTSP5.  Only then do we clear the way for new
documentation.

LTSP19 will likely not be the last version.  We should understand and plan
for this.  Hence, moving new documentation under this.

The new documentation that needs to be written is a massive project on its
own.  This includes:

- Writing basic how-to's designed for new users.
  - Installation (needs to be made new user friendly)
  - Printer integration (setting CUPS to use the printers on the server, I
also have to look at the use of local printers to see if I can coax cups
into doing what I want it to)
  - Scanner integration (sane.d)
  - Setting up LTSP for the Raspberry Pie
  - Setting up LTSP for other project board computers
  - Setting up vnc to the client

- Extended how-tos
  - Writeups of per computer options in ltsp.conf and dhcp.conf
  - Multiple OS options (have to figure out if it can serve a Windows
image...)
  - Setting up clusters
  - Setting up retro pie
  - Setting up student-teacher software
  - User management documentation
  - Domain integration
  - All of the man pages need to be turned into functional how-tos
  - Much of the current documentation needs to be turned into functional
how-tos and guides for regular users
  - Much of the WIKI stuff and bug reports needs to be turned into
functional documentation.

Then I will need to comb everything to flesh out other options that will
need documentation.

This is a 3 month project if I was working on it full time.  As a
when-I-have-time hobby, it may take me 2 years.

**Comment by alkisg** [comment]
Hi Steven,

oh I wasn't complaining about the timing at all. It's a **lot** better to have someone caring about documentation slowly over 3 years, than have someone write some documentation in a month and then leave. I'm sure that the LTSP community will greatly appreciate your efforts there.

I was just pointing this: I do want the deprecated UbuntuLTSP wiki to stay deprecated; I do not want to participate in any effort to revive it, even as a side-effect. For me it's very wrong to maintain distro-specific documentation when upstream tries to be cross-distro, and additionally to maintain documentation in 2 places when upstream is missing documentation contributors. I feel that reviving the UbuntuLTSP wiki will harm LTSP users.

So all I was asking is "let's delete/revert the UbuntuLTSP wiki and maintain documentation in https://github.com/ltsp/community/wiki". If we can agree there, we can cooperate, otherwise each one of us will just do it his own way.

Thanks!

**Comment by alkisg** [comment]
A month ago I had a similar discussion with @rkwesk for the Debian wiki:

https://github.com/ltsp/community/issues/29

After 10+ years of participating in various LTSP documentation efforts, I'm personally convinced that at this point, trying to maintain distro-specific wikis will just harm LTSP users. Of course anyone may feel otherwise and just go ahead and do it; but I don't want to help in something that will harm LTSP users...

There were many, many cases where LTSP users came in IRC and asked for help, saying "I followed that wiki page or that tutorial and LTSP doesn't work", and I had to waste time reading it just to tell them that "it's completely broken, please don't follow that"; while in the LTSP community wiki, I could keep an eye for really broken instructions...

**Comment by stevensantos-docs** [comment]
For me, the Ubuntu pages will be a first draft, useful in letting clients see a writing process. The Ubuntu help pages have proven to be helpful in getting me paid writing work, where lesser used documentation sites don't seem to count for nearly as much with my paying clients.  That said, I do get your rational.

I also see a major issue in leaving the Ubuntu help/wiki alone.  It's a liability, as its hugely referenced, and done from the main help pages.  Right now, I would bet that many of those wiki pages get more views in a day than the upstream does in a month.   From a documentation perspective, THIS IS A PROBLEM, and one that won't go away by ignoring it.   I don't see a good rationale for leaving these very high-volume pages in disrepair at a time when we want more people using the software.  Most new users (not power users, well, maybe some of them) will hit the Ubuntu help pages, specifically issue pages and stop when it doesn't work.  Thats not good for this project.  In a cleanup of the type I am doing, adding in a header to the old pages directing them to the LTSP19 stuff is fairly trivial.  

My server is Ubuntu.  The reality is that this is where I will start.  Then it will need to be generalized out.  Thats a LOT easier (at least for me) when you have already written it to one architecture to then go and generalize it out and find the differences you need to document.  This will become even more important when an RPM based LTSP19 distro spins up. 

The IRC issue isn't going to die.  Right now, I am willing to maintain both the LTSP one and the Ubuntu one.  Yes, I need to do a LOT more to cleanup the Ubuntu one before I can write much new documentation, but its something I am willing to do.  At some point the benefits of the Ubuntu help pages will be less (for example, the sane documents don't do much for me any more), and then contributing to it less important.  But I have specific reasons for needing relatively recent things on the Ubuntu wiki.

**Comment by alkisg** [comment]
> Right now, I would bet that many of those wiki pages get more views in a day than the upstream does in a month.

Then by replacing them all with just a link to our wiki, it will move Google's pagerank from those sites to our site. This is a win-win solution. In the ltsp wiki you'll be one of the main documentation persons, listed in the respective github team (hence getting work via that), while in the UbuntuLTSP wiki you'll just be a downstream maintainer, that will have to explain to his clients "why the official LTSP documentation mentions that the UbuntuLTSP documentation is completely unsupported by upstream developers?"

Also, git is a whole lot easier to work with than both moinmoin and mediawiki. You can clone it locally, do whatever you want, edit, rename directories, move files around etc, and upload. Or back it up, or move it to gitlab etc. We're admins there; in UbuntuLTSP we're just low-level contributors.

**Comment by stevensantos-docs** [comment]
Look, at the end of the day, their is a significant real-world benefit to
me publishing help documents on the Ubuntu help/wiki.  My clients and
potential clients read that.  They are impressed by the Ubuntu name. The
github name is less impressive to them (power users, not developers).  The
LTSP name by itself won't get them excited to hire me.

Longer term, when Redhat spins up a good LTSP implementation, when other
distros start really supporting LTSP, the equation changes.  Then you have
these other projects cloning the LTSP documentation and that means a higher
distribution in other places that also matter.  That's why I am willing to
do both.

**Comment by alkisg** [comment]
From 2008 to 2013 we were writing content in the UbuntuLTSP wiki.
Since 2014 or so we deprecated it and pointed it to wiki.ltsp.org (that's why wiki.ltsp.org comes before UbuntuLTSP in Google results).
People stopped reading the UbuntuLTSP wiki and they started reading wiki.ltsp.org.
Since 2016 or so wiki.ltsp.org is unmaintained so we're moving to github.

I don't know why you **now** feel that LTSP users are still reading the UbuntuLTSP wiki and you want to **start** investing there; but I think we both clearly stated our goals and we agreed that we don't want to cooperate on this.

I'm going to rename this issue to "Discussion about documentation hosting", move it to community issues, and I'll open another one for Raspberries.

Cheers,
Alkis

**Comment by stevensantos-docs** [comment]
Alkis,

You have made it clear that my choosing to include the Ubuntu help documents in my efforts is not acceptable to you.  

I will finish what reorg I started on the Ubuntu help, but you have made it clear that documenting the new LTSP project is not a project that is worthwhile to me. 

Hopefully you will find someone else who will be willing to do the documentation.  

I wish you luck in your project.

**Comment by alkisg** [comment]
Steven you may do whatever you want in the UbuntuLTSP wiki; I just don't want to *personally* waste time on something that I feel will harm LTSP users.

But please do rename all those LTSP19 pages; the old name is "LTSP5" and the new is "LTSP"; writing LTSP19 there next year will be extremely confusing as the version then will be LTSP 20.
There's no point in putting the version number of a software in the URLs, except for LTSP5 which is obsolete and will never have a version change again.
